src/pip/_internal/utils/filesystem.py

```diff
@@ -161,7 +161,7 @@ def _test_writable_dir_win(path):
         except OSError as e:
             if e.errno == errno.EEXIST:
                 continue
-            if e.errno == errno.EPERM:
+            if e.errno == errno.EPERM or e.errno == errno.EACCES:
                 # This could be because there's a directory with the same name.
                 # But it's highly unlikely there's a directory called that,
                 # so we'll assume it's because the parent dir is not writable.
```

Comment:
You can update the comment as well indicating the other reason, this would return `False`.